File: NM2/lab08/app/analytical.py
import numpy as np

def analytial(x_range, y_range, t_range, h_x, h_y, tau, solution):
    x = np.arange(*x_range, h_x)
    y = np.arange(*y_range, h_y)
    t = np.arange(*t_range, tau)

    res = np.zeros((len(t), len(x), len(y)))
    for i in range(len(x)):
        for j in range(len(y)):
            for k in range(len(t)):
                res[k][i][j] = solution(x[i], y[j], t[k])
    
    return res

File: NM2/lab06/internal/analytical/analytical.go
package analytical

import (
	"lab06/internal/config"
	"lab06/pkg/tools"
	"math"
)

func Analytical(c config.Config, solution func(x, t, a float64) float64) [][]float64 {
	tau := math.Sqrt(c.Sigma * math.Pow(c.H, 2))
	x := tools.Arange(c.XBegin, c.XEnd, c.H)
	t := tools.Arange(c.TBegin, c.TEnd, tau)

	res := tools.Zeros(len(t), len(x))
	for i := range x {
		for j := range t {
			res[j][i] = solution(x[i], t[j], c.A)
		}
	}
	return res
}
